x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java

```diff
@@ -167,6 +178,7 @@ void performDuringNoSnapshot(IndexMetadata indexMetadata, ClusterState currentCl
         if (totalShardsPerNode != null) {
             settingsBuilder.put(ShardsLimitAllocationDecider.INDEX_TOTAL_SHARDS_PER_NODE_SETTING.getKey(), totalShardsPerNode);
         }
+        settingsBuilder.put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, numberOfReplicas);
```

Comment:
As I mentioned above, we should make this optional and only set if specified by the policy